diff --git a/config.def.h b/config.def.h
index 91ab8ca..8d0772f 100644
--- a/config.def.h
+++ b/config.def.h
@@ -143,6 +143,9 @@ static unsigned int defaultrcs = 257;
  */
 static unsigned int cursorshape = 2;

+/* Switch to decide blinking or not */
+static int cursorblinks = 1;
+
 /*
  * Default columns and rows numbers
  */
diff --git a/x.c b/x.c
index 2a3bd38..a161eb6 100644
--- a/x.c
+++ b/x.c
@@ -1564,10 +1564,14 @@ xdrawcursor(int cx, int cy, Glyph g, int ox, int oy, Glyph og)
 		case 0: /* Blinking Block */
 		case 1: /* Blinking Block (Default) */
 		case 2: /* Steady Block */
+			if (IS_SET(MODE_BLINK))
+				break;
 			xdrawglyph(g, cx, cy);
 			break;
 		case 3: /* Blinking Underline */
 		case 4: /* Steady Underline */
+			if (IS_SET(MODE_BLINK))
+				break;
 			XftDrawRect(xw.draw, &drawcol,
 					borderpx + cx * win.cw,
 					borderpx + (cy + 1) * win.ch - \
@@ -1576,6 +1580,8 @@ xdrawcursor(int cx, int cy, Glyph g, int ox, int oy, Glyph og)
 			break;
 		case 5: /* Blinking bar */
 		case 6: /* Steady bar */
+			if (IS_SET(MODE_BLINK))
+				break;
 			XftDrawRect(xw.draw, &drawcol,
 					borderpx + cx * win.cw,
 					borderpx + cy * win.ch,
@@ -1983,6 +1989,10 @@ run(void)
 		if (FD_ISSET(ttyfd, &rfd) || xev) {
 			if (!drawing) {
 				trigger = now;
+				if (IS_SET(MODE_BLINK)) {
+					win.mode ^= MODE_BLINK;
+				}
+				lastblink = now;
 				drawing = 1;
 			}
 			timeout = (maxlatency - TIMEDIFF(now, trigger)) \
@@ -1993,7 +2003,7 @@ run(void)

 		/* idle detected or maxlatency exhausted -> draw */
 		timeout = -1;
-		if (blinktimeout && tattrset(ATTR_BLINK)) {
+		if (blinktimeout && (cursorblinks || tattrset(ATTR_BLINK))) {
 			timeout = blinktimeout - TIMEDIFF(now, lastblink);
 			if (timeout <= 0) {
 				if (-timeout > blinktimeout) /* start visible */
